thunder/torch/__init__.py

```diff
@@ -170,23 +170,18 @@ def is_floating_point(a: TensorLike, /) -> bool:
 
 
 # Handles the size method
-def size(a):
-    def fn_(idx: int | None = None):
-        if idx is None:
-            return a.shape
-        return a.shape[idx]
-
-    return fn_
+@torchsymbol(torch.Tensor.size, is_method=True, id="torch.Tensor.size")
```

Comment:
This PR is a good overall direction to go in. I think the issue is that we've been too aggressive at labeling some functions symbols.

Could we update this function, `is_floating_point`, and `is_cuda` to not be symbols? Updating all of them may be beyond the scope of this PR, so maybe this should just modify the changes to `size` so it's not a symbol.

I think we don't want to make this a symbol because it's just extracting the shape from a tensor, which is currently constexpr (so the symbol is constexpr, and doesn't need to be evaluated), and in the future will just return an existing proxy, anyway.

fyi @jjsjann123, @IvanYashchuk and @nikitaved on this thinking

Reply:
Fixed